File: 100-doors/doors.go
import "fmt"

func main() {
  fmt.Println("100 doors")
  steps := 1 
  doors := [100]bool{}
  while(steps <= 100) {
    for i := 1; i <=100; i += steps {
      doors[i] = !doors[i]      
    }
    steps++;
  }

  fmt.Println(doors)
}

File: 100-doors/doors.rs
fn main() {
    let doors: [bool; 100] = [false; 100];
    let steps: u16 = 1;
    while steps <= 100 {
        for door in doors.iter() {
            *door = !*door;
        }
        steps += 1;
    }

    println!("The doors are {:?}", doors);
}
